src/Equinox.MessageDb/MessageDbClient.fs

```diff
@@ -14,15 +14,24 @@ type private Format = ReadOnlyMemory<byte>
 [<Struct>]
 type ExpectedVersion = Any | StreamVersion of int64
 
+module private Queries =
+    [<Literal>]
+    let writeMessage = "select * from write_message($1::text, $2, $3, $4, $5, $6)"
+    [<Literal>]
+    let readLast = "select position, type, data, metadata, id::uuid, time from get_last_stream_message($1, $2);"
+    [<Literal>]
+    let readStream = "select position, type, data, metadata, id::uuid, time from get_stream_messages($1, $2, $3);"
+
+
 module private Sql =
-    let addExpectedVersion name (value: ExpectedVersion) (p: NpgsqlParameterCollection) =
+    let addExpectedVersion (value: ExpectedVersion) (p: NpgsqlParameterCollection) =
```

Comment:
having this off over here doesnt add much value - why not do a one-liner match to illustrate the value mapping, and then do a normal AddWithValue inline (maybe with a `let inline t v (p: ParamCol) -> p.AddWithValue(t, v) |> ignore)

Reply:
> why not do a one-liner match to illustrate the value mapping, and then do a normal AddWithValue inline

```fsharp
let expectedVersionToBigInt v = match v with StreamVersion v -> v | Any -> DbNull
```

^ I don't think this would compile

Reply:
yes, needs a box (see my push, but doesnt matter)